src/molusce/moluscedialog.py

```diff
@@ -474,6 +586,62 @@ def checkGeometry(self):
         self.geometry_matched = True
         self.__updateAnalyticTabs(self.geometry_matched)
 
+    def checkConsistencySeparateVars(self):
+        # separate spatial variables for simulations should be:
+        # - same number of variables as training ones
+        # - same number of bands for each matching variable
+        # - match geometry of initial rasters and training variables
+        if not utils.checkFactors(self.inputs, sim=True):
+            QMessageBox.warning(
+                self,
+                self.tr("Missed input data"),
+                self.tr(
+                    "Factors rasters is not set. Please specify them and try again"
+                ),
+            )
+            return
+
+        if not (len(self.inputs['factors']) == len(self.inputs['factors_sim'])):
+            QMessageBox.warning(
+                    self,
+                    self.tr("Different number of variables"),
+                    self.tr(
+                        "Model is trained using {} variables, and simulation was set up with {} variables"
+                    ).format(len(self.inputs['factors']),len(self.inputs['factors_sim'])),
+                )
+            return
+
+        initRaster = self.inputs["initial"]
+        for (_k, v), (_k2, v2) in zip(self.inputs["factors_sim"].items(), self.inputs["factors"].items()):
```

Comment:
For zipping values you can use `zip(self.inputs["factors_sim"].values(), self.inputs["factors"].values())`